src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateParts = date.split(separatorFrom);
+  const newDate = {};
+
+  for (let i = 0; i < dateParts.length; i++) {
+    const key = fromFormat[i];
+    const value = dateParts[i];
+
+    newDate[key] = value;
+  }
+
+  const resultDate = toFormat
+    .map((e) => {
+      if (!newDate[e]) {
+        if (e === 'YY') {
+          return newDate.YYYY.slice(-2);
+        } else {
```

Comment:
This condition checks if `newDate[e]` does not exist, which is only possible for 'YY' and 'YYYY' conversions. It would be clearer to check explicitly for 'YYYY' to 'YY' and 'YY' to 'YYYY' conversions to improve readability and maintainability of the code.